### ISSUE TITLE
Stuck on "Starting continue server" on Windows 11

### ISSUE THREAD

**Issue by All-Ki** [continuedev/continue#300]
The extension is stuck on "Starting continue server" since around 10 minutes. I have python 3.10 as my python version.
Also tried launching vscode as Admin, didn't help

**Comment by d-imal** [comment]
I'm in a similar situation. It eventually stopped showing that message, but the docs say that the extension should open on its own and bring up a Python file. That didn't happen, so I tried just working with one of my own files. None of the commands return anything. The README says to open dev tools and check the logs to see if the server is running, so I did that. There's nothing in the logs, so I assume it's not running, but there are no instructions on how to start it. I'm on Python 3.8.4 / MacOS.

**Comment by d-imal** [comment]
Nevermind. I was on a VPN and I forgot that OpenAI blocks my VPN. 

Getting a different error now, though. I might open a different issue for that.

**Comment by sestinj** [comment]
@All-Ki are there any additional details you could provide about the situation? If anything it would greatly help to know whether you are on MacOS/Linux or Windows. Working on this and will get back to you with a fix

**Comment by AymericG** [comment]
I can't start the server, it says "**No module named 'continuedev'**"

![image](https://github.com/continuedev/continue/assets/446402/a2ecfb8d-b156-49f3-9e91-9b260d0b05a7)

**Comment by bhaktatejas922** [comment]
having issues as well. extension needs a lot of work still but looks promising

**Comment by simseb** [comment]
I'm on Ubuntu 22.04, python 3.10.6 and the extension stays at 'Loading Continue server...'. The server has started, but it responds with **404**; (i also entered my API key)
![Selection_139](https://github.com/continuedev/continue/assets/36695238/1c1747f2-993e-41c7-a12b-40eb2a67d512)
![Selection_140](https://github.com/continuedev/continue/assets/36695238/a47f2064-392d-4d31-9955-6514fd85cfc5)

Maybe it helps... can't wait to use it :)

**Comment by tom-voicer** [comment]
> I can't start the server, it says "**No module named 'continuedev'**"
> 
> ![image](https://user-images.githubusercontent.com/446402/256424332-a2ecfb8d-b156-49f3-9e91-9b260d0b05a7.png)

i have the exact same problem.. any fixes?

**Comment by cyril-schroeder-ox** [comment]
> 

Pretty much the same case, Ubuntu 22.04, Python 3.10.6: "Loading Continue server..." with the arrow slowly blinking but nothing seems to further happen.

**Comment by All-Ki** [comment]
@sestinj 
Sorry, i'm on windows.
Any additionnal info you'd like ? And i don't know how to bring up the logs with VSCode, but if you tell me i can send those to you

**Comment by ferrantejake** [comment]
Added a relevant ticket with slightly different error details here: https://github.com/continuedev/continue/issues/311

**Comment by bhaktatejas922** [comment]
@TyDunn it seems like theres a lot of dependency issues. I dont know much about vs code extensions but maybe it'd help to package it as a binary using pyinstaller instead of installing packages locally?

**Comment by bhaktatejas922** [comment]
<img width="547" alt="image" src="https://github.com/continuedev/continue/assets/26863466/ae154351-ff4a-4df5-ba5e-1a044f035dd3">
Stuck on loading Continue Server

**Comment by d-imal** [comment]
After (maybe) resolving my other issues, I'm stuck here again. I think some feedback in the UI would be helpful. I think the python packages are done downloading and it's hanging for some other reason, but there's nothing in the UI to tell you what's going on. What's weird is that I checked `~/.continue/continue.log` and I see this:

```
INFO:     Started server process [91858]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:65432 (Press CTRL+C to quit)
INFO:     ('127.0.0.1', 54842) - "WebSocket /ide/ws?session_id=a7003805-15de-4cf6-a6ba-edb4690cc7a6" [accepted]
INFO:     connection open
```
But the "Starting server" message is up and no commands work...

**Comment by sestinj** [comment]
For everyone who was experiencing this issue, there have been numerous updates that probably solve the problem, as well as improvements to logging which will quickly offer more information if something goes wrong. I'm going to wait to hear back from you all on whether it is fixed, but would like to close this issue and move to more specific issues once possible (we've already done this in a few cases). Thanks!

**Comment by dcstrange** [comment]
![tmp1920](https://github.com/continuedev/continue/assets/19701927/b7de6dac-5f68-4d3e-b01b-5327f32b15fc)

**Comment by sestinj** [comment]
@dcstrange Thanks for sending over the console logs. This problem is one we know how to solve, and you can read a bit more [here](https://continue.dev/docs/troubleshooting#for-windows-users). I've also pushed changes that will address that `TypeError` you are seeing.

**Comment by sestinj** [comment]
And for everyone in this thread still experiencing the issue, I've created much better [troubleshooting documentation here](https://continue.dev/docs/troubleshooting), but MORE IMPORTANTLY and working on a permanent solution: we will soon ship with a packaged executable instead of trying to install python packages on your machine. You can track progress [here](https://github.com/continuedev/continue/issues/320), and I'll comment here as well once it is complete.

**Comment by sestinj** [comment]
Closing this issue. We have since
1. Packaged the extension into a reliable binary with pyinstaller
2. Endured a week+ of testing without hearing these exact problems
3. Included testing as a part of CI to ensure that the server can start

Anyone can feel free to reopen if the issue is persisting and we'll address it right away
